src/ast/passes/codegen_llvm.cpp

```diff
@@ -3458,31 +3535,14 @@ void CodegenLLVM::generate_maps(const RequiredResources &resources)
 
   // bpftrace internal maps
   for (const StackType &stack_type : resources.stackid_maps) {
-    // bpf_get_stackid() is kind of broken by design.
-    //
-    // First, stacks can suffer from hash collisions and we can lose stacks. We
-    // receive collision errors in userspace b/c we do not set
-    // BPF_F_REUSE_STACKID. Note that we should **NEVER** set this flag b/c then
-    // we can silently receive unrelated stacks in userspace.
-    //
-    // Second, there is not a great way to reduce the frequency of collision
-    // (other than increasing map size) b/c multiple probes can be sharing the
-    // same stack (remember they are hashed). As a result, we cannot delete
-    // entries in this map b/c we don't know when all the sharers are gone.
-    //
-    // Fortunately, we haven't seen many bug reports about missing stacks (or
-    // maybe people don't care that much). The proper solution would be to use
-    // bpf_get_stack() to get the actual stack trace and pass it to userspace
-    // through the ring buffer.  However, there is additional complexity with
-    // this b/c we need to set up a percpu map to use as scratch space for
-    // bpf_get_stack() to write into. Using the stack directly wouldn't work
-    // well b/c it would take too much stack space.
-    //
-    // The temporary fix is to bump the map size to 128K. Any further bumps
-    // should warrant consideration of the previous paragraph.
     createMapDefinition(stack_type.name(),
-                        libbpf::BPF_MAP_TYPE_STACK_TRACE,
+                        libbpf::BPF_MAP_TYPE_HASH,
```

Comment:
i forget, what happens if we manage to fill up the map? do we start dropping entries?

Reply:
I'm thinking we still cannot clear a stack from the map after reading, cuz the same stack could've been captured again in kernelspace.

Reply:
> i forget, what happens if we manage to fill up the map? do we start dropping entries?

Yes, I believe so. I think we already have some automatic logging if map update fails e.g.
https://github.com/bpftrace/bpftrace/blob/master/src/ast/irbuilderbpf.cpp#L473